vhdbuilder/packer/test/linux-vhd-content-test.sh

```diff
@@ -117,21 +117,36 @@ validateDownloadPackage() {
 }
 
 validateOrasOCIArtifact() {
-  local downloadURL=$1
+  local referenceByTag=$1
   local downloadedPackage=$2
-  echo "Validating package $downloadURL from registry and downloaded package $downloadedPackage"
+  local testDescription=$3
+  echo "Validating package $referenceByTag from registry and downloaded package $downloadedPackage"
+
+  # Fetch the manifest and extract the size and digest using jq
+  manifest_json=$(oras manifest fetch --registry-config ${ORAS_REGISTRY_CONFIG_FILE} "$referenceByTag")
+  fileSizeInRegistry=$(echo "$manifest_json" | jq '.layers[0].size')
+  layerDigestInRegistry=$(echo "$manifest_json" | jq -r '.layers[0].digest')
 
-  # Fetch the manifest and extract the size using jq
-  fileSizeInRegistry=$(oras manifest fetch --registry-config ${ORAS_REGISTRY_CONFIG_FILE} "$downloadURL" | jq '.layers[0].size')
-  
   # Get the size of the downloaded package
   fileSizeDownloaded=$(wc -c "$downloadedPackage" | awk '{print $1}' | tr -d '\r')
-  
+
   # Compare the sizes
   if [ "$fileSizeInRegistry" != "$fileSizeDownloaded" ]; then
-    echo "Error: File size mismatch. Expected $fileSizeInRegistry, but got $fileSizeDownloaded."
+    err $testDescription "File size of ${downloadedPackage} from ${referenceByTag} is invalid. Expected file size: ${fileSizeInRegistry} - downloaded file size: ${fileSizeDownloaded}"
     return 1
   fi
+  # Compare the digests
+  if [ "${layerDigestInRegistry#sha}" != "$layerDigestInRegistry" ]; then
+    # Only check digest if it starts with 'sha', blake3 not supported
+    algo_and_digest=${layerDigestInRegistry%%:*}
+    digest_value=${layerDigestInRegistry#*:}
+    algo=${algo_and_digest#sha}
+    computedDigest=$(shasum -a "$algo" "$downloadedPackage" | awk '{print $1}')
```

Comment:
shasum doesnt exists on azlinux3 it's making the build fail.

Reply:
I have updated the tool used to algorithm specific ones, e.g. sha256sum, which should be installed as GNU core utils.

Also added handling to skip digest verification when no tools are available. PTAL